### PR TITLE
[INT-200] Check database connection on startup and on `/healthz` endpoint

### PR DESCRIPTION
**Pull request by archessmn**
Do database connection monitoring slightly better

### CODE REVIEW

File: server/lib.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    while (connectionAttempts <= 3) {
+      try {
+        await prisma.$executeRaw(Prisma.sql`SELECT 1;`);
+        return resolve();
+      } catch (e) {
+        console.error(
+          `Database connection attempt ${connectionAttempts} failed, retrying...`,
+        );
+        await sleep(5000);
```
Comment (markspolakovs) [inline server/lib.ts:24]: Should we bail immediately and not do the whole retry loop / is this delaying the inevitable?

File: app/healthz/route.ts [hunk 2]
```diff
@@ -1,7 +1,13 @@
+import { dbHealthCheck } from "@/features/lib";
```
Comment (markspolakovs) [inline app/healthz/route.ts:1]: That's a slightly grim import path - feel free to break the "no importing from lib in app" rule for this file and silence the eslint warning.

File: app/healthz/route.ts [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
 
 export const dynamic = "force-dynamic";
 
-export function GET() {
-  return new NextResponse("ok");
+export async function GET() {
+  try {
+    await dbHealthCheck();
+  } catch (e) {
+    return new NextResponse("not ok :(", { status: 500 });
```
Comment (markspolakovs) [inline app/healthz/route.ts:10]: ¿porqué?

File: server/lib.ts (hunk 1 above)
Comment (archessmn) [reply server/lib.ts:24]: My reasoning was just to give it a bit of time to get started so it wouldn't just fail instantly and keep getting redeployed if it had an issue on the first connection

File: app/healthz/route.ts (hunk 2 above)
Comment (archessmn) [reply app/healthz/route.ts:1]: I tried that but it's an error not a warning, although changing the import path should probably be done

File: server/lib.ts (hunk 1 above)
Comment (markspolakovs) [reply server/lib.ts:24]: Possibly? But in my head that'd only be the case if the *database server* is the one that's taking a while to start, which it should almost never be in production (the Prisma connection should block if it isn't ready yet), so we'd want to fail fast if something is up with the DB.

File: app/healthz/route.ts (hunk 3 above)
Comment (archessmn) [reply app/healthz/route.ts:10]: If it's not connected to the db, it's not going to work at all and so monitoring should reflect that

File: app/healthz/route.ts (hunk 2 above)
Comment (markspolakovs) [reply app/healthz/route.ts:1]: Did a `// eslint-disable-next-line @typescript-eslint/no-restricted-imports` before it not work? That's odd

Comment (archessmn) [reply app/healthz/route.ts:1]: I didn't try that tbh, I'll have a look when I get home later

File: app/healthz/route.ts (hunk 3 above)
Comment (markspolakovs) [reply app/healthz/route.ts:10]: sorry, what I meant is "the response should tell you porqué it no work" :p

File: server/lib.ts (hunk 1 above)
Comment (archessmn) [reply server/lib.ts:24]: It's within 15 seconds, it's not going to cause a massive issue I don't think

**Pull request merged**